driver/registry_base.go

```diff
@@ -102,11 +104,26 @@ func (m *RegistryBase) OAuth2AwareMiddleware() func(h http.Handler) http.Handler
 	return m.oa2mw
 }
 
+func (m *RegistryBase) AddCORSonHandler(h http.Handler) http.Handler {
+	c := cors.New(cors.Options{
+		AllowedMethods:   []string{"GET", "POST", "PUT", "PATCH", "DELETE"},
+		AllowedHeaders:   []string{"Authorization", "Content-Type"},
+		ExposedHeaders:   []string{"Content-Type"},
+		AllowCredentials: true,
+	})
+	return c.Handler(h)
+}
+
 func (m *RegistryBase) RegisterRoutes(admin *x.RouterAdmin, public *x.RouterPublic) {
 	m.HealthHandler().SetHealthRoutes(admin.Router, true)
 	m.HealthHandler().SetVersionRoutes(admin.Router)
 
-	m.HealthHandler().SetHealthRoutes(public.Router, false)
+	// Register a healthcheck setup
+	hcHandlers := health.HealthCheck{}
+	hcHandlers.RegisterHealthHandlers(m.HealthHandler(), false)
+
+	public.Handler("GET", healthx.AliveCheckPath, m.AddCORSonHandler(hcHandlers.Alive))
+	public.Handler("GET", healthx.ReadyCheckPath, m.AddCORSonHandler(hcHandlers.Ready))
```

Comment:
I could be wrong, but i don't think we need to recreate the healthx routes inside hydra since we have them defined in ory/x and it is being reused across all projects.

I think the best approach here is to wrap the health handler with the cors middleware.
We retrieve the health handler inside the `registry_base` so we should be able to wrap them here.

https://github.com/ory/hydra/blob/3115dde229a6be936ad4d844d778d6ee82279643/driver/registry_base.go#L217-L241

Reply:
I guess @barelyhuman did this because

> I've intentionally not added it for the version route from the `healthx` library.

But I would also say we just wrap the whole health handler including the version.

Reply:
The thing is that the `cors` library needs a proper `http.Handler` reference and the `healthx` one's are built on top of httprouter's handler signature, also why I had to create the `HealthCheck` type, the type only wraps the healthx calls. 

I could get rid of the wrapper but then we'd need to either enable it on the all the routes of the router or add a new router that handles it and the new router would need to be on the same iface which complicates understanding what's going on

Reply:
i see, okay maybe let's rather then have a parameter on the healthx package that accepts the `router` and a `handler` middleware. Then in ory/x we do the wrapping of these paths if the middleware is defined 
https://github.com/ory/x/blob/master/healthx/handler.go#L84-L87

We can then just create the middleware inside hydra and then pass it along to ory/x.
I think this would be the best since we then don't duplicate code and don't need to refactor a lot

Reply:
https://github.com/ory/hydra/issues/2989#issuecomment-1128606791

back to the original set of thoughts then, 
cool, I'll raise a PR the healthx with the modification and also change this one accordingly

Reply:
Sorry, we did not expect it to be that complicated here. You can first do the PR in `ory/x` and then adjust this one as needed once we merged it.

Reply:
It's not a biggie, will probably take this up post work today 👍🏼

Reply:
Just out of interest I looked into the `healthx` package a bit and I can see it is poorly written. I imagined it would have an http.Handler return type so you could just wrap it in your middleware.

For a quick and dirty solution one could probably do something like this 

```go
opt, _ := m.C.CORS(config.PublicInterface)
c := cors.New(opt)

public.Handler("GET", healthx.AliveCheckPath, c.Handler(http.HandlerFunc(func(writer http.ResponseWriter, request *http.Request) {
   m.HealthHandler().Alive(writer, request, nil)
})))

public.Handler("GET", healthx.ReadyCheckPath, c.Handler(http.HandlerFunc(func(writer http.ResponseWriter, request *http.Request) {
   m.HealthHandler().Ready(false)(writer, request, nil)
})))
```

but yeah i think it's best to fix the healthx package :laughing:

Reply:
> Just out of interest I looked into the `healthx` package a bit and I can see it is poorly written. I imagined it would have an `http.Handler` return type so you could just wrap it in your middleware.
> 
> For a quick and dirty solution one could probably do something like this
> 
> ```go
> opt, _ := m.C.CORS(config.PublicInterface)
> c := cors.New(opt)
> 
> public.Handler("GET", healthx.AliveCheckPath, c.Handler(http.HandlerFunc(func(writer http.ResponseWriter, request *http.Request) {
>    m.HealthHandler().Alive(writer, request, nil)
> })))
> 
> public.Handler("GET", healthx.ReadyCheckPath, c.Handler(http.HandlerFunc(func(writer http.ResponseWriter, request *http.Request) {
>    m.HealthHandler().Ready(false)(writer, request, nil)
> })))
> ```
> 
> but yeah i think it's best to fix the healthx package 😆

I did do this, then I moved it to match with the New + Composition pattern being used everywhere and ended up refactoring to use a new struct altogether 

Though, as long as the repos using healthx are version locked , i think the changes made there should be fine. 
Wouldn't wanna break everything

Reply:
yes they are version locked :)